src/main/java/ch/njol/skript/config/Node.java

```diff
@@ -422,9 +420,77 @@ public String toString() {
 			+ (comment.isEmpty() ? "" : " " + comment)
 			+ " (" + config.getFileName() + ", " + (lineNum == -1 ? "unknown line" : "line " + lineNum) + ")";
 	}
-	
+
 	public boolean debug() {
 		return debug;
 	}
-	
+
+	/**
+	 * @return The index of this node relative to the other children of this node's parent,
+	 * or -1 if this node does not have a parent. The index includes counted void nodes.
+	 */
+	int getIndex() {
+		if (parent == null)
+			return -1;
+
+		int index = 0;
+		for (Iterator<Node> iterator = parent.fullIterator(); iterator.hasNext(); ) {
+			Node node = iterator.next();
+			if (node == this)
+				return index;
+
+			index++;
+		}
+		return -1;
+	}
+
+	/**
+	 * Returns the path to this node in the config file from the root.
+	 *
+	 * <p>
+	 * Getting the path of node {@code z} in the following example would
+	 * return an array with {@code w.x, y, z}.
+	 * <pre>
+	 *     w.x:
+	 *      y:
+	 *       z: true # this node
+	 * </pre></p>
+	 *
+	 * @return The path to this node in the config file.
+	 */
+	public @NotNull String[] getPath() {
+		List<String> path = new ArrayList<>();
+		Node node = this;
+
+		while (node != null) {
+			if (node.getKey() == null || node.getKey().isEmpty())
+				break;
+
+			path.add(0, node.getKey() + ".");
+			node = node.getParent();
+		}
+
+		if (path.isEmpty())
+			return new String[0];
+
+		int lastIndex = path.size() - 1;
+		String lastValue = path.get(lastIndex);
+		path.set(lastIndex, lastValue.substring(0, lastValue.length() - 1)); // trim trailing dot
+		return path.toArray(new String[0]);
+	}
+
+	@Override
+	public boolean equals(Object object) {
+		if (!(object instanceof Node other))
+			return false;
+
+		return Arrays.equals(getPath(), other.getPath()) // for entry/section nodes
+			&& Objects.equals(comment, other.comment); // for void nodes
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hash(Arrays.hashCode(getPath()), comment);
```

Comment:
Is there a reason you're calculating the path hashcode too? Rather than just:
```suggestion
		return Objects.hash(getPath(), comment);
```

Reply:
gives a warning in intellij that getPath should be wrapped in Arrays.hashCode

Reply:
fair enough